client/src/components/Chat.jsx

```diff
@@ -482,7 +483,7 @@ const Chat = () => {
 												}`}
 											>
 												{containsBadword && !isSender && !badwordChoices[id] ? (
-													<div className="flex flex-col border-red border w-full rounded-r-md p-3">
+													<div className={`${!settings.theme && 'badword-lightmode-text-background'} flex flex-col border-red border w-full rounded-r-md p-3`}>
```

Comment:
Here you just need to set the proper styles for the dark mode, using the prefix and then the regular styles it should have otherwise.

Example: `dark:text-white text-black`